### DISCUSSION TITLE
Extend `debug/gdscript/warnings/exclude_addons` to allow exclusion of custom paths.

### DISCUSSION THREAD

**Discussion by Jack-023** [godotengine/godot-proposals#10099] [Editor]
It would be helpful if we could add a list of files/directories to exclude, rather than only excluding the addons directory. Currently I am using [imjp94/gd-plug](https://github.com/imjp94/gd-plug) to manage plugins in my project, which keeps a copy of the full git repo in `res://.plugged` to aid in managing versions of plugins.

This causes the issue of triggering warnings for all plugins since the source code is now also in a directory other than `res://addons`. This is especially problematic when using non-default settings for the LSP warnings. For example I have enabled warnings/errors for `untyped_declarations`, `unsafe_property_access`, `unsafe_method_access`, `unsafe_cast`, and `return_value_discarded` which most plugins don't adhere to causing even a few installed plugins to generate hundreds of warnings. In the mean time I am deleting the `res://.plugged` dir every time I install/update plugins but this is not ideal.

**Comment by dalexeev** [comment]
* Duplicate of #9146.
* See also godotengine/godot/pull/93889.

Thanks for the contribution nonetheless!

**Reply by Jack-023** [reply]
Ah I thought I had seen that before but when I tried searching today I couldn't find it. Sorry for reposting. Thanks for getting on to this, hope your change gets merged soon.
